# Task 2

## CUSTOMER RETENTION & CHURN ANALYSIS

🔹**Task**: Analyze customer data to identify churn patterns, key retention drivers,
and customer lifetime trends for a subscription-based business.

🔹**Skills Gained**: Retention analysis, cohort analysis, customer lifetime metrics,
insight-driven decision making

🔹**Deliverable**: A retention analysis dashboard or report highlighting churn reasons,
retention trends, and actionable recommendations to reduce customer loss.

**1️⃣ Project Overview**

Dataset name : telco_customer_churn.csv

**Business objective :**

To analyze customer retention and churn behavior in a subscription-based business by identifying churn patterns, key retention drivers, customer lifetime trends, and revenue impact in order to generate actionable strategies that reduce customer loss and improve long-term profitability.

**Key questions I will answer:**

1. What churn patterns can be identified within the customer base?

2. Which factors act as key retention drivers in the subscription-based business?

3. What are the customer lifetime trends across different customer segments?

4. How does churn impact overall customer lifetime and revenue performance?

5. What insights can be derived to reduce customer loss and improve long-term retention?

**🔎 Dataset Description**

The dataset contains 7,043 customer records across 21 features.

It represents customer subscription details, service usage, billing information, and churn status for a telecom-based subscription business.

**Features overview :** 

customerID – Unique customer identifier

gender – Customer gender

SeniorCitizen – Whether the customer is a senior citizen (1 = Yes, 0 = No)

Partner – Whether the customer has a partner

Dependents – Whether the customer has dependents

tenure – Number of months the customer has stayed with the company

PhoneService – Whether the customer has phone service

MultipleLines – Whether the customer has multiple lines

InternetService – Type of internet service (DSL, Fiber optic, No)

OnlineSecurity – Whether the customer has online security

OnlineBackup – Whether the customer has online backup

DeviceProtection – Whether the customer has device protection

TechSupport – Whether the customer has tech support

StreamingTV – Whether the customer uses streaming TV

StreamingMovies – Whether the customer uses streaming movies

Contract – Contract type (Month-to-month, One year, Two year)

PaperlessBilling – Whether billing is paperless

PaymentMethod – Customer payment method

MonthlyCharges – Monthly subscription charge

TotalCharges – Total amount charged to the customer

Churn – Whether the customer left the company (Yes/No)


In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
o_telco_customer_data = pd.read_csv("telco_customer_churn.csv")
o_telco_customer_data


,customerID,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,...,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,MonthlyCharges,TotalCharges,Churn
0,7590-VHVEG,Female,0,Yes,No,1,No,No phone service,DSL,No,...,No,No,No,No,Month-to-month,Yes,Electronic check,29.85,29.85,No
1,5575-GNVDE,Male,0,No,No,34,Yes,No,DSL,Yes,...,Yes,No,No,No,One year,No,Mailed check,56.95,1889.5,No
2,3668-QPYBK,Male,0,No,No,2,Yes,No,DSL,Yes,...,No,No,No,No,Month-to-month,Yes,Mailed check,53.85,108.15,Yes
3,7795-CFOCW,Male,0,No,No,45,No,No phone service,DSL,Yes,...,Yes,Yes,No,No,One year,No,Bank transfer (automatic),42.30,1840.75,No
4,9237-HQITU,Female,0,No,No,2,Yes,No,Fiber optic,No,...,No,No,No,No,Month-to-month,Yes,Electronic check,70.70,151.65,Yes
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
7038,6840-RESVB,Male,0,Yes,Yes,24,Yes,Yes,DSL,Yes,...,Yes,Yes,Yes,Yes,One year,Yes,Mailed check,84.80,1990.5,No
7039,2234-XADUH,Female,0,Yes,Yes,72,Yes,Yes,Fiber optic,No,...,Yes,No,Yes,Yes,One year,Yes,Credit card (automatic),103.20,7362.9,No
7040,4801-JZAZL,Female,0,Yes,Yes,11,No,No phone service,DSL,Yes,...,No,No,No,No,Month-to-month,Yes,Electronic check,29.60,346.45,No
7041,8361-LTMKD,Male,1,Yes,No,4,Yes,Yes,Fiber optic,No,...,No,No,No,No,Month-to-month,Yes,Mailed check,74.40,306.6,Yes



### Made a copy of the original dataset for the data cleaning process.

In [3]:
telco_customer_data = o_telco_customer_data.copy()
telco_customer_data.head()

,customerID,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,...,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,MonthlyCharges,TotalCharges,Churn
0,7590-VHVEG,Female,0,Yes,No,1,No,No phone service,DSL,No,...,No,No,No,No,Month-to-month,Yes,Electronic check,29.85,29.85,No
1,5575-GNVDE,Male,0,No,No,34,Yes,No,DSL,Yes,...,Yes,No,No,No,One year,No,Mailed check,56.95,1889.5,No
2,3668-QPYBK,Male,0,No,No,2,Yes,No,DSL,Yes,...,No,No,No,No,Month-to-month,Yes,Mailed check,53.85,108.15,Yes
3,7795-CFOCW,Male,0,No,No,45,No,No phone service,DSL,Yes,...,Yes,Yes,No,No,One year,No,Bank transfer (automatic),42.30,1840.75,No
4,9237-HQITU,Female,0,No,No,2,Yes,No,Fiber optic,No,...,No,No,No,No,Month-to-month,Yes,Electronic check,70.70,151.65,Yes


In [5]:
telco_customer_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7043 non-null   object 
 1   gender            7043 non-null   object 
 2   SeniorCitizen     7043 non-null   int64  
 3   Partner           7043 non-null   object 
 4   Dependents        7043 non-null   object 
 5   tenure            7043 non-null   int64  
 6   PhoneService      7043 non-null   object 
 7   MultipleLines     7043 non-null   object 
 8   InternetService   7043 non-null   object 
 9   OnlineSecurity    7043 non-null   object 
 10  OnlineBackup      7043 non-null   object 
 11  DeviceProtection  7043 non-null   object 
 12  TechSupport       7043 non-null   object 
 13  StreamingTV       7043 non-null   object 
 14  StreamingMovies   7043 non-null   object 
 15  Contract          7043 non-null   object 
 16  PaperlessBilling  7043 non-null   object 


In [6]:
telco_customer_data.describe()

,SeniorCitizen,tenure,MonthlyCharges
count,7043.000000,7043.000000,7043.000000
mean,0.162147,32.371149,64.761692
std,0.368612,24.559481,30.090047
min,0.000000,0.000000,18.250000
25%,0.000000,9.000000,35.500000
50%,0.000000,29.000000,70.350000
75%,0.000000,55.000000,89.850000
max,1.000000,72.000000,118.750000


In [7]:
telco_customer_data.describe(include = 'object')

,customerID,gender,Partner,Dependents,PhoneService,MultipleLines,InternetService,OnlineSecurity,OnlineBackup,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,TotalCharges,Churn
count,7043,7043,7043,7043,7043,7043,7043,7043,7043,7043,7043,7043,7043,7043,7043,7043,7043,7043
unique,7043,2,2,2,2,3,3,3,3,3,3,3,3,3,2,4,6531,2
top,3186-AJIEK,Male,No,No,Yes,No,Fiber optic,No,No,No,No,No,No,Month-to-month,Yes,Electronic check,20.2,No
freq,1,3555,3641,4933,6361,3390,3096,3498,3088,3095,3473,2810,2785,3875,4171,2365,11,5174


In [13]:
telco_customer_data.duplicated().sum()

np.int64(0)

## STEP 1 : DATA CLEANING

**Step 1.1 : Remove/handle missing values (TotalCharges)**

In [14]:
telco_customer_data['TotalCharges'].isnull().sum()

np.int64(0)

**Step 1.2 : Convert data types (tenure → int, charges → float)**

In [15]:
telco_customer_data.dtypes

customerID           object
gender               object
SeniorCitizen         int64
Partner              object
Dependents           object
tenure                int64
PhoneService         object
MultipleLines        object
InternetService      object
OnlineSecurity       object
OnlineBackup         object
DeviceProtection     object
TechSupport          object
StreamingTV          object
StreamingMovies      object
Contract             object
PaperlessBilling     object
PaymentMethod        object
MonthlyCharges      float64
TotalCharges         object
Churn                object
dtype: object

In [16]:
telco_customer_data['TotalCharges'] = pd.to_numeric(
    telco_customer_data['TotalCharges'], 
    errors='coerce'
)

In [17]:
telco_customer_data.dtypes

customerID           object
gender               object
SeniorCitizen         int64
Partner              object
Dependents           object
tenure                int64
PhoneService         object
MultipleLines        object
InternetService      object
OnlineSecurity       object
OnlineBackup         object
DeviceProtection     object
TechSupport          object
StreamingTV          object
StreamingMovies      object
Contract             object
PaperlessBilling     object
PaymentMethod        object
MonthlyCharges      float64
TotalCharges        float64
Churn                object
dtype: object

**Step : 1.3 From the feature TotalChages Drop NaN**

In [18]:
telco_customer_data[telco_customer_data['TotalCharges'].isna()]

,customerID,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,...,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,MonthlyCharges,TotalCharges,Churn
488,4472-LVYGI,Female,0,Yes,Yes,0,No,No phone service,DSL,Yes,...,Yes,Yes,Yes,No,Two year,Yes,Bank transfer (automatic),52.55,NaN,No
753,3115-CZMZD,Male,0,No,Yes,0,Yes,No,No,No internet service,...,No internet service,No internet service,No internet service,No internet service,Two year,No,Mailed check,20.25,NaN,No
936,5709-LVOEQ,Female,0,Yes,Yes,0,Yes,No,DSL,Yes,...,Yes,No,Yes,Yes,Two year,No,Mailed check,80.85,NaN,No
1082,4367-NUYAO,Male,0,Yes,Yes,0,Yes,Yes,No,No internet service,...,No internet service,No internet service,No internet service,No internet service,Two year,No,Mailed check,25.75,NaN,No
1340,1371-DWPAZ,Female,0,Yes,Yes,0,No,No phone service,DSL,Yes,...,Yes,Yes,Yes,No,Two year,No,Credit card (automatic),56.05,NaN,No
3331,7644-OMVMY,Male,0,Yes,Yes,0,Yes,No,No,No internet service,...,No internet service,No internet service,No internet service,No internet service,Two year,No,Mailed check,19.85,NaN,No
3826,3213-VVOLG,Male,0,Yes,Yes,0,Yes,Yes,No,No internet service,...,No internet service,No internet service,No internet service,No internet service,Two year,No,Mailed check,25.35,NaN,No
4380,2520-SGTTA,Female,0,Yes,Yes,0,Yes,No,No,No internet service,...,No internet service,No internet service,No internet service,No internet service,Two year,No,Mailed check,20.00,NaN,No
5218,2923-ARZLG,Male,0,Yes,Yes,0,Yes,No,No,No internet service,...,No internet service,No internet service,No internet service,No internet service,One year,Yes,Mailed check,19.70,NaN,No
6670,4075-WKNIU,Female,0,Yes,Yes,0,Yes,Yes,DSL,No,...,Yes,Yes,Yes,No,Two year,No,Mailed check,73.35,NaN,No


**11 rows are NaN in the feature TotalCharges**

In [19]:
# Drop the NaN from TotalCharges
telco_customer_data.dropna(subset=['TotalCharges'], inplace=True)

In [20]:
telco_customer_data

,customerID,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,...,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,MonthlyCharges,TotalCharges,Churn
0,7590-VHVEG,Female,0,Yes,No,1,No,No phone service,DSL,No,...,No,No,No,No,Month-to-month,Yes,Electronic check,29.85,29.85,No
1,5575-GNVDE,Male,0,No,No,34,Yes,No,DSL,Yes,...,Yes,No,No,No,One year,No,Mailed check,56.95,1889.50,No
2,3668-QPYBK,Male,0,No,No,2,Yes,No,DSL,Yes,...,No,No,No,No,Month-to-month,Yes,Mailed check,53.85,108.15,Yes
3,7795-CFOCW,Male,0,No,No,45,No,No phone service,DSL,Yes,...,Yes,Yes,No,No,One year,No,Bank transfer (automatic),42.30,1840.75,No
4,9237-HQITU,Female,0,No,No,2,Yes,No,Fiber optic,No,...,No,No,No,No,Month-to-month,Yes,Electronic check,70.70,151.65,Yes
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
7038,6840-RESVB,Male,0,Yes,Yes,24,Yes,Yes,DSL,Yes,...,Yes,Yes,Yes,Yes,One year,Yes,Mailed check,84.80,1990.50,No
7039,2234-XADUH,Female,0,Yes,Yes,72,Yes,Yes,Fiber optic,No,...,Yes,No,Yes,Yes,One year,Yes,Credit card (automatic),103.20,7362.90,No
7040,4801-JZAZL,Female,0,Yes,Yes,11,No,No phone service,DSL,Yes,...,No,No,No,No,Month-to-month,Yes,Electronic check,29.60,346.45,No
7041,8361-LTMKD,Male,1,Yes,No,4,Yes,Yes,Fiber optic,No,...,No,No,No,No,Month-to-month,Yes,Mailed check,74.40,306.60,Yes


**Step 1.4 : Convert Churn to binary (1/0)**

In [22]:
telco_customer_data['Churn'] = (
    telco_customer_data['Churn']
    .replace({'Yes': 1, 'No': 0})
    .infer_objects(copy=False)
)

In [23]:
telco_customer_data['Churn'].head()

0    0
1    0
2    1
3    0
4    1
Name: Churn, dtype: int64

## STEP 2 : BUSINESS METRICS

**Step 2.1 : Calculating Overall Churn Rate**

In [25]:
# Churn Rate = Customers who left / Total customers.
churn_rate = sum(telco_customer_data['Churn']) / len(telco_customer_data['Churn'])
(churn_rate * 100)

26.578498293515356

**Identified an overall churn rate of 26.5%, indicating approximately 1 in 4 customers discontinued service.**

**Step 2.2 : Calculating Retention Rate**

In [26]:
# Retention Rate = 1 − Churn Rate
retention_rate = 1 - churn_rate
(retention_rate * 100)

73.42150170648465

**Identified an overall retention rate of 73.5%, indicating that nearly 3 out of 4 customers continued their subscription with the company.**

**Step 2.3 : Calculating Avg Tenure**

In [27]:
avg_tenure = telco_customer_data['tenure'].mean()
avg_tenure

np.float64(32.421786120591584)

**The typical customer remains subscribed for roughly 2 years and 8 months.**

**Step 2.4 : Caculating Avg Monthly Charges**

In [28]:
avg_monthly_charges = telco_customer_data['MonthlyCharges'].mean()
avg_monthly_charges

np.float64(64.79820819112628)

**On average, customers pay approximately $64.79 per month.**

**Step 2.5 : Calculating Avg Lifetime Value (MonthlyCharges × tenure)**

In [29]:
avg_lifetime_value = telco_customer_data['MonthlyCharges'] * telco_customer_data['tenure']
avg_lifetime_value

0         29.85
1       1936.30
2        107.70
3       1903.50
4        141.40
         ...   
7038    2035.20
7039    7430.40
7040     325.60
7041     297.60
7042    6972.90
Length: 7032, dtype: float64

In [30]:
avg_lifetime_value.mean()

np.float64(2283.147248293515)

**This Means :**

**On average:**

**A customer stays ≈ 32 months**

**Pays ≈ $64.76 per month**

**Generates ≈ $2,279 in total revenue during their lifetime**

**Step 2.6 : Calculating Customer Count by Contract**

In [31]:
telco_customer_data['Contract'].unique()

array(['Month-to-month', 'One year', 'Two year'], dtype=object)

In [32]:
telco_customer_data['Contract'].value_counts()

Contract
Month-to-month    3875
Two year          1685
One year          1472
Name: count, dtype: int64

**55% of customers are on month-to-month contracts, showing a strong preference for flexibility and a potentially higher churn risk compared to long-term contracts.**

**Step 2.7 : Calculating Revenue by Churn vs Non-Churn**

In [33]:
revenue_by_churn = telco_customer_data.groupby('Churn')['TotalCharges'].sum()
revenue_by_churn

Churn
0    13193241.8
1     2862926.9
Name: TotalCharges, dtype: float64

**The majority of revenue comes from retained customers, while churned customers contributed significantly less. Reducing churn could help protect and expand long-term revenue streams.**

## STEP 3 : DEEP CHURN ANALYSIS

**Step 3.1 : Churn by Contract Type**

In [34]:
churn_by_contract = pd.crosstab(
    telco_customer_data['Contract'],
    telco_customer_data['Churn'],
    normalize='index'
) * 100

churn_by_contract

Churn,0,1
Contract,,
Month-to-month,57.290323,42.709677
One year,88.722826,11.277174
Two year,97.151335,2.848665


**Month-to-month customers have extremely high churn**

**Two-year contracts almost never churn**

**STRONG BUSINESS INSIGHT:**

**Contract length is the strongest predictor of churn. Encouraging long-term contracts could significantly reduce churn rate.**

**Step 3.2 : Churn by Internet Service**

In [35]:
churn_by_internet_service = pd.crosstab(
    telco_customer_data['InternetService'],
    telco_customer_data['Churn'],
    normalize='index'
) * 100
churn_by_internet_service

Churn,0,1
InternetService,,
DSL,81.001656,18.998344
Fiber optic,58.107235,41.892765
No,92.565789,7.434211


**Customers subscribed to fiber optic services exhibit the highest churn rate (~42%), nearly double that of DSL users and significantly higher than customers without internet service. This suggests pricing sensitivity or service dissatisfaction within the fiber segment, making it a critical focus area for churn reduction strategies.**

**Step 3.3 Churn by Payment Method**

In [36]:
churn_by_payment_method = pd.crosstab(
    telco_customer_data['PaymentMethod'],
    telco_customer_data['Churn'],
    normalize='index'
) * 100
churn_by_payment_method

Churn,0,1
PaymentMethod,,
Bank transfer (automatic),83.268482,16.731518
Credit card (automatic),84.746877,15.253123
Electronic check,54.714588,45.285412
Mailed check,80.798005,19.201995


**Customers using electronic check show higher churn rates, possibly indicating lower engagement or automatic billing adoption issues.**

**Step 3.4 Churn by Senior Citizen**

In [37]:
churn_by_senior_citizen = pd.crosstab(
    telco_customer_data['SeniorCitizen'],
    telco_customer_data['Churn'],
    normalize='index'
) * 100
churn_by_senior_citizen

Churn,0,1
SeniorCitizen,,
0,76.349745,23.650255
1,58.318739,41.681261


**Senior customers may require better service support or tailored plans.**

**Step 3.5 Churn by Tenure Buckets**

In [38]:
telco_customer_data['TenureGroup'] = pd.cut(
    telco_customer_data['tenure'],
    bins=[0,12,24,48,60,72],
    labels=['0-12','12-24','24-48','48-60','60-72']
)

In [39]:
telco_customer_data.head()

,customerID,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,...,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,MonthlyCharges,TotalCharges,Churn,TenureGroup
0,7590-VHVEG,Female,0,Yes,No,1,No,No phone service,DSL,No,...,No,No,No,Month-to-month,Yes,Electronic check,29.85,29.85,0,0-12
1,5575-GNVDE,Male,0,No,No,34,Yes,No,DSL,Yes,...,No,No,No,One year,No,Mailed check,56.95,1889.50,0,24-48
2,3668-QPYBK,Male,0,No,No,2,Yes,No,DSL,Yes,...,No,No,No,Month-to-month,Yes,Mailed check,53.85,108.15,1,0-12
3,7795-CFOCW,Male,0,No,No,45,No,No phone service,DSL,Yes,...,Yes,No,No,One year,No,Bank transfer (automatic),42.30,1840.75,0,24-48
4,9237-HQITU,Female,0,No,No,2,Yes,No,Fiber optic,No,...,No,No,No,Month-to-month,Yes,Electronic check,70.70,151.65,1,0-12


In [40]:
churn_by_tenure_group = pd.crosstab(
    telco_customer_data['TenureGroup'],
    telco_customer_data['Churn'],
    normalize='index'
) * 100
churn_by_tenure_group

Churn,0,1
TenureGroup,,
0-12,52.321839,47.678161
12-24,71.289062,28.710938
24-48,79.611041,20.388959
48-60,85.576923,14.423077
60-72,93.390192,6.609808


**Most churn happens in the first year of subscription. Early-stage retention strategies are critical.**

**Step 3.6 Churn by Monthly Charges Buckets**

In [41]:
telco_customer_data['ChargesGroup'] = pd.cut(
    telco_customer_data['MonthlyCharges'],
    bins=[0,35,70,100,120],
    labels=['Low','Medium','High','Very High']
)

churn_by_monthly_charges_bucket = pd.crosstab(
    telco_customer_data['ChargesGroup'],
    telco_customer_data['Churn'],
    normalize='index'
) * 100
churn_by_monthly_charges_bucket

Churn,0,1
ChargesGroup,,
Low,89.068826,10.931174
Medium,76.016260,23.983740
High,62.150056,37.849944
Very High,71.951220,28.048780


In [43]:
telco_customer_data.head()

,customerID,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,...,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,MonthlyCharges,TotalCharges,Churn,TenureGroup,ChargesGroup
0,7590-VHVEG,Female,0,Yes,No,1,No,No phone service,DSL,No,...,No,No,Month-to-month,Yes,Electronic check,29.85,29.85,0,0-12,Low
1,5575-GNVDE,Male,0,No,No,34,Yes,No,DSL,Yes,...,No,No,One year,No,Mailed check,56.95,1889.50,0,24-48,Medium
2,3668-QPYBK,Male,0,No,No,2,Yes,No,DSL,Yes,...,No,No,Month-to-month,Yes,Mailed check,53.85,108.15,1,0-12,Medium
3,7795-CFOCW,Male,0,No,No,45,No,No phone service,DSL,Yes,...,No,No,One year,No,Bank transfer (automatic),42.30,1840.75,0,24-48,Medium
4,9237-HQITU,Female,0,No,No,2,Yes,No,Fiber optic,No,...,No,No,Month-to-month,Yes,Electronic check,70.70,151.65,1,0-12,High


**Higher charges → higher churn**

**Customers paying premium prices may be more price-sensitive or expect higher service quality.**

## STEP 4: RETENTION & RISK INSIGHTS

**Step 4.1 : Churn by Tenure (Early Churn Detection)**

In [46]:
# Create Tenure Buckets
telco_customer_data['TenureGroup'] = pd.cut(
    telco_customer_data['tenure'],
    bins=[0,12,24,36,48,60,72],
    labels=['0-12','12-24','24-36','36-48','48-60','60-72']
)
# Calculate Churn % by Tenure Group
churn_by_tenure = pd.crosstab(
    telco_customer_data['TenureGroup'],
    telco_customer_data['Churn'],
    normalize='index'
) * 100

churn_by_tenure


Churn,0,1
TenureGroup,,
0-12,52.321839,47.678161
12-24,71.289062,28.710938
24-36,78.365385,21.634615
36-48,80.971129,19.028871
48-60,85.576923,14.423077
60-72,93.390192,6.609808


**Nearly half of customers churn within the first year, while long-term customers show very low churn, making early retention efforts critical.**

**Step 4.2 : High-Value Customer Churn**

In [50]:
# Create CLV (Customer Lifetime Value)
telco_customer_data['CLV'] = (
    telco_customer_data['MonthlyCharges'] *
    telco_customer_data['tenure']
)
# Compare Average CLV by Churn
telco_customer_data.groupby('Churn')['CLV'].mean()
# Revenue at Risk (Total Revenue from Churned Customers)
telco_customer_data.loc[
    telco_customer_data['Churn'] == 1,
    'TotalCharges'
].sum()


np.float64(2862926.9)

**Churned customers account for $2.86M in lost revenue, highlighting significant financial impact and the importance of reducing churn.**

**Step 4.3 : Long-Term vs Short-Term Customer Stability**


In [52]:
# Create Stability Category
telco_customer_data['CustomerType'] = telco_customer_data['tenure'].apply(
    lambda x: 'Short-Term' if x <= 24 else 'Long-Term'
)
# Churn % by Customer Type
pd.crosstab(
    telco_customer_data['CustomerType'],
    telco_customer_data['Churn'],
    normalize='index'
) * 100

Churn,0,1
CustomerType,,
Long-Term,85.963997,14.036003
Short-Term,58.393248,41.606752


**Short-term customers are nearly 3x more likely to churn than long-term customers, making early retention a critical business priority.**

## Step 5 : Export Clean Data

In [53]:
# Exporting the Clean Dataset
telco_customer_data.to_csv("cleaned_dataset.csv",index =False)